sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -966,6 +967,22 @@ def _update_clustering_key(
             # ref: https://cloud.google.com/bigquery/docs/creating-clustered-tables#modifying-cluster-spec
             self.execute(exp.update(table_name, {c: c for c in cluster_by}, where=exp.true()))
 
+    @contextlib.contextmanager
+    def temp_table(
+        self,
+        query_or_df: QueryOrDF,
+        name: TableName = "diff",
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        **kwargs: t.Any,
+    ) -> t.Iterator[exp.Table]:
+        # ensure that we use default catalog if no catalog is not specified else it would use the execution project as default
+        if isinstance(name, exp.Table) and not name.catalog:
```

Comment:
what happens if name is not a table? should probably call to_table()

Reply:
should this be done in the base class, and not just bigquery?

Reply:
Yeah, I cant think of a reason not to do this in the base class. It would make all references fully qualified / unambiguous

Reply:
`name` is a `TableName` which means it's either a `exp.Table` or a `str`. Then if it's a `str`, I assume it's either a `table` (in which case, it could be relevant but I'm not sure in which case it is?) or `catalog.db.table` which means that it might not be relevant to override the provided catalog by the default one.
However I see what you mean by expecting the `to_table()` wrapping but I wasn't it wouldn't mess up with something else.

So would you the following workflow:
- if name is an str => convert it to table
- Check if that exp.table has a catalog
- if not add the default one else keep existing one
- call super implementation

 works for you?

Reply:
yep thanks, and let's move this logic to the base class?

Reply:
Oh ok then I'll move it back to the base class 👍

Reply:
I wanted to use: 
```
        if isinstance(name, str):
            name = exp.to_table(name)
        # ensure that we use default catalog if none is not specified
        if not name.catalog:
            name.set("catalog", value=self.default_catalog)
```

However as you can see it leads to issues to:
- postgres (and I don't really understand why but it fails 1 test with the catalog change and 2 tests if I add the to_table)
- spark (I'm not sure why it changes the name of the table when I set the catalog?)
- table_diff.test_generated_sql (I'm not sure how it changes from a CREATE to a DROP 🤯)

I tried with only: 
```
 if not name.catalog:
            name.set("catalog", value=self.default_catalog)
```

but it's definitely that part that leads to most regressions. 

Do you really want to try to explore that approach? It looks like much harder to fix that way though.